tests/wpunit/PostObjectQueriesTest.php

```diff
@@ -1532,4 +1532,74 @@ public function testGraphQLQueryShortcodeInContent() {
 
 	}
 
+	/**
+	 * Test post can be queried its permalink
+	 */
+	public function testPostByPermalinkQuery() {
+		$post_id = $this->createPostObject( [
+			'post_type' => 'post',
+			'post_title' => 'Post title',
+		] );
+
+		$permalink = get_permalink($post_id);
```

Comment:
@epeli I believe you said this _should_ also work with the relative path? If so can we get a test for that as well to make sure it continues to work in the future?

Also, naming is hard. Is `permalink` a good name or is that a "WordPress" term? 🤔

Reply:
I was thinking that test like that just tests the get_permalink() implementation.

Yeah, `permalink` is bit WordPressish. I chose it because the `url_to_postid` documentation says 

> $url  (string) (Required) Permalink to check.

And the current graphql api exposes a permalink field so I think it nicely analogous with it.

Reply:
@epeli on the PostObjects we expose it as `link`

We are fairly inconsistent here though. Some places we do `url`, some we do `sourceUrl` and here we do `link`

🤔

---

Did I mention that naming is hard? 😂 

---

Maybe for consistency with how it's exposed on the PostObjects, have it supported here as `link` instead of `permalink`?

Reply:
Damn I'm pretty sure I saw permalink in somewhere in the graphql apis...

But if so `link` is better indeed.

Reply:
ya, it's `link` here: https://github.com/wp-graphql/wp-graphql/blob/79d2c4d0f2093553d94718962398863b5c7c3db8/src/Type/Object/PostObject.php#L386

Reply:
It's now a `link`.